posthog/queries/properties_timeline/properties_timeline.py

```diff
@@ -0,0 +1,96 @@
+import json
+from typing import Any, Dict, List, TypedDict
+
+from posthog.models.filters.properties_timeline_filter import PropertiesTimelineFilter
+from posthog.models.person.person import Person
+from posthog.models.property.util import extract_tables_and_properties, get_single_or_multi_property_string_expr
+from posthog.models.team.team import Team
+from posthog.queries.insight import insight_sync_execute
+
+from .properties_timeline_event_query import PropertiesTimelineEventQuery
+
+
+class PropertiesTimelinePoint(TypedDict):
+    timestamp: str
+    properties: Dict[str, Any]
+    relevant_events_since_previous_point: int
+
+
+PROPERTIES_TIMELINE_SQL = """
+SELECT
+    timestamp,
+    properties,
+    if(
+        NOT is_pre_range AND start_event_number = 1,
+        0, /* If the event is first-ever for this person, relevant_events_since_previous_point will be 0 */
+        start_event_number - previous_start_event_number
+    ) AS relevant_events_since_previous_point
+FROM (
+    SELECT
+        timestamp,
+        properties,
+        is_pre_range,
+        start_event_number,
+        lagInFrame(start_event_number) OVER person_points AS previous_start_event_number
+    FROM (
+        SELECT
+            timestamp,
+            person_properties AS properties,
+            is_pre_range,
+            {crucial_property_columns} AS relevant_property_values, -- TODO make this dynamic
+            lagInFrame({crucial_property_columns}) OVER person_events AS previous_relevant_property_values,
+            row_number() OVER person_events AS start_event_number
+        FROM ({event_query})
```

Comment:
Flat is usually better than nested (see https://github.com/PostHog/posthog/issues/13315). Could we ditch the event query somehow?

Reply:
Not really because window functions. At best `relevant_property_values` could be directly in the event query, but from my testing this performs about the same as the current version.

Reply:
I think we can drop this, but I'll take issue with

> performs about the same as the current version.

This is unclear without additional details. 

https://github.com/PostHog/posthog/issues/13315 Trends with nested event queries also performs identically to one where nesting is removed _until_ you have too much data and your network suddenly isn't fast enough. The indicator whether something is wrong here is network traffic for the subquery sent to the other shard.

Also note testing on team_id=2 is useless for performance optimizations - we should be looking at top 10 US clients as we're too small to be meaningful. Also also - we should be controlling for page cache :)

All this said - as long as we do proper benchmarking on this before enabling this endpoint for everyone (feature flagging the FE), I think we might be good. Happy to pair on the benchmarking of this next week.

Reply:
Karl, all of these queries filter on events for a *single* person within a given time range, so even if we were just getting all rows from Clickhouse and doing aggregation in python, I think we'd be reasonably fast 😅 

(hence why I also dropped my suggestion of using arrays / going too deep in the weeds to figure out how that would work)